webapp/channels/src/components/sidebar_right/sidebar_right.tsx

```diff
@@ -131,6 +137,55 @@ export default class SidebarRight extends React.PureComponent<Props, State> {
         }
     };
 
+    handleRHSFocus(prevProps: Props) {
+        const wasOpen = prevProps.isOpen;
+        const isOpen = this.props.isOpen;
+
+        const contentChanged = (
+            (this.props.isPinnedPosts !== prevProps.isPinnedPosts) ||
+            (this.props.isRecentMentions !== prevProps.isRecentMentions) ||
+            (this.props.isSavedPosts !== prevProps.isSavedPosts) ||
+            (this.props.isChannelFiles !== prevProps.isChannelFiles) ||
+            (this.props.isChannelInfo !== prevProps.isChannelInfo) ||
+            (this.props.isChannelMembers !== prevProps.isChannelMembers) ||
+            (this.props.isPostEditHistory !== prevProps.isPostEditHistory) ||
+            (this.props.rhsChannel?.id !== prevProps.rhsChannel?.id) ||
+            (this.props.teamId !== prevProps.teamId)
+        );
+
+        if (this.props.isOpen && (contentChanged || (!wasOpen && isOpen))) {
+            this.previousActiveElement = document.activeElement as HTMLElement;
+            requestAnimationFrame(() => {
+                if (this.sidebarRight.current) {
+                    document.dispatchEvent(
+                        new CustomEvent<A11yFocusEventDetail>(A11yCustomEventTypes.FOCUS, {
+                            detail: {
+                                target: this.sidebarRight.current,
+                                keyboardOnly: false,
+                            },
+                        }),
+                    );
+                }
+            });
+        } else if (!this.props.isOpen && wasOpen) {
+            // RHS just was closed, restore focus to the previous element had it
+            // this will have to change for upcoming work specially for search and probalby plugins
```

Comment:
well, you are right that the RHS allows you to move around even when opened, but I still find valuable the option that when closing you get back to the previous point where you were. I was playing with the focus on other places, and still find valuable to be able to move (while RHS open) and when deciding to close the RHS via focus, getting back to the place that initially triggered the open.
And in the case that you choose other element that triggers the opening of the RHS, then the last element would be that one and the focus will be back to it (I tried opening recent mentions, then moving around until channel members, then close the RHS and the focus got back to channel members button). 
And lastly, when switching teams, RHS is closed and the focus is restarted to the opened channel post box, so no problem in that scenario then.

Reply:
We discussed more of this in a DM. I'm still thinking not sure about refocusing the previous element when the RHS is closed, but since LevelAccess is treating the RHS as a dialog, this behaviour is needed for that